src/main/java/io/vertx/gsoc2023/qotd/QoTDHandlers.java

```diff
@@ -23,36 +34,52 @@ public void getAllQuotes(RoutingContext ctx) {
         .end(quotes.encode()));
   }
 
+  public void postNewQuote(RoutingContext context) {
+    var payload = context.body().asJsonObject();
+    insertQuote(payload)
+      .onFailure(cause -> context.response().setStatusCode(400).end())
+      .onSuccess(quote -> {
+        eventBus.send(QUOTE_INSERTED, quote);
+        context.response().setStatusCode(201).end();
+      });
+  }
+
+  public void realtimeQuotes(RoutingContext context) {
+    context.request()
+      .toWebSocket()
+      .onSuccess(wsServer -> {
+        quoteInsertedConsumer.handler(message -> {
+          wsServer.writeBinaryMessage(message.body().toBuffer());
+        });
+      });
+  }
+
   private Future<JsonArray> fetchAllQuotes() {
     return pool
       .query("SELECT * FROM quotes")
       .execute()
-      .compose(rs -> {
+      .map(rs -> {
         var jsonArray = new JsonArray();
-        rs.forEach(row -> {
-          var quote = new JsonObject()
-            .put("quoteId", row.getLong("quote_id"))
-            .put("text", row.getString("text"))
-            .put("author", row.getString("author"));
-          jsonArray.add(quote);
-        });
-        return Future.succeededFuture(jsonArray);
+        rs.forEach(row -> jsonArray.add(row.toJson()));
+        return jsonArray;
       });
   }
 
-  public void postNewQuote(RoutingContext context) {
-    var payload = context.body().asJsonObject();
-    insertQuote(payload)
-      .onFailure(cause -> context.response().setStatusCode(400).end())
-      .onSuccess(__ -> context.response().setStatusCode(201).end());
-  }
-
-  private Future<Void> insertQuote(JsonObject quote) {
+  private Future<JsonObject> insertQuote(JsonObject quote) {
     if (quote.getString("text") == null)
       return failedFuture(new IllegalArgumentException("The 'text' field must be provided"));
     return pool
-      .preparedQuery("INSERT INTO quotes (text, author) VALUES ($1, $2)")
-      .execute(Tuple.of(quote.getString("text"), quote.getString("author", "Unknown")))
-      .mapEmpty();
+      .getConnection()
+      .compose(conn ->
+        conn
+          .preparedQuery("INSERT INTO quotes (text, author) VALUES ($1, $2) RETURNING quote_id")
+          .execute(Tuple.of(quote.getString("text"), quote.getString("author", "Unknown")))
+          .map(rows -> rows.iterator().next().getLong("quote_id"))
+          .compose(quoteId -> conn
+            .preparedQuery("SELECT * FROM quotes WHERE quote_id=$1")
+            .execute(Tuple.of(quoteId))
+            .map(rows -> rows.iterator().next().toJson()))
```

Comment:
Postgres allows to return more columns than the generated id. You could create the json object with a single query.

Reply:
This was in fact a great tip I will take for life.